hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -299,9 +301,25 @@ checkClose ctx headContext parties initialUtxoHash snapshotNumber closedUtxoHash
 -- | Checks that the datum
 makeContestationDeadline :: ContestationPeriod -> ScriptContext -> POSIXTime
 makeContestationDeadline cperiod ScriptContext{scriptContextTxInfo} =
-  case ivTo (txInfoValidRange scriptContextTxInfo) of
-    UpperBound (Finite time) _ -> addContestationPeriod time cperiod
-    _ -> traceError "no upper bound validaty interval defined for close"
+  let txValidFrom = ivFrom (txInfoValidRange scriptContextTxInfo)
+      txValidTo = ivTo (txInfoValidRange scriptContextTxInfo)
```

Comment:
I'd suggest to use a top-down approach in validators. It makes reading them much easier.